### ISSUE TITLE
A parent calling refresh due to a child calling dispatchEvent in a schedule callback causes stack overflow

### ISSUE THREAD

**Issue by brainkim** [bikeshaving/crank#166]
```jsx
function *Child() {
  this.schedule(() => {
    this.dispatchEvent(new Event("test", {bubbles: true}));
  });

  for ({} of this) {
    yield <div>Child</div>;
  }
}

function Parent() {
  this.addEventListener("test", () => {
    this.refresh();
  });

  return <Child />;
}
```

For some reason, the refresh triggers a new execution of `Child`, rather than re-executing the old one. Likely we need to change the order of when `schedule` executes or something. Alternatively, maybe we should forbid `dispatchEvent ` from being called in a `schedule` callback.

If you encounter a bug like this, as a temporary workaround, try putting the `dispatchEvent` call in a `setTimeout`.

**Comment by brainkim** [comment]
I’m having trouble reproducing this in the test suite. Need to investigate further.

**Comment by brainkim** [comment]
Same underlying issue as https://github.com/bikeshaving/crank/issues/216 and should be fixed when that issue is fixed.

**Comment by brainkim** [comment]
Fixed in 0.4.0 (probably)
